File: ben/haskell/Q21-28-lists-again/p25.hs
-- Generate a random permutation of the elements of a list.

import System.Random
import Data.List

rnd_permu :: [a] -> IO [a]
rnd_permu list = do g <- getStdGen
                    let r = head (randomRs (0, length list - 1) g)
                        in return $ (permutations list)!!r

File: ben/prolog/P01-P28-lists/p25.pl
% Generate a random permutation of the elements of a list.

rnd_permu([], []).
rnd_permu(L, [X|R]) :-
    length(L, Len),
    Len > 0, 
    random(0, Len, I),
    nth0(I, L, X, Xs),
    rnd_permu(Xs, R).
